ui/components/app/nft-details/nft-details.tsx

```diff
@@ -0,0 +1,861 @@
+                  frameTextTitleProps={{
+                    textAlign: TextAlign.Center,
+                    color: TextColor.textAlternative,
+                    variant: TextVariant.bodyMdMedium,
+                  }}
+                  frameTextTitleStyle={{
+                    fontSize: '10px',
+                    lineHeight: '16px',
+                  }}
+                  value={
+                    lastSale?.price?.amount?.usd
+                      ? formatCurrency(
+                          `${lastSale?.price?.amount?.usd}`,
+                          currency,
+                          getPricePrecision(lastSale?.price?.amount?.usd),
+                        )
+                      : t('dataUnavailable')
+                  }
+                  frameTextValueProps={{
+                    color: hasLastSalePrice
+                      ? TextColor.textDefault
+                      : TextColor.textAlternative,
+                    variant: hasLastSalePrice
+                      ? TextVariant.headingSm
+                      : TextVariant.bodyMdMedium,
+                    textAlign:
+                      hasLastSalePrice && lastSale?.orderSource
+                        ? undefined
+                        : TextAlign.Center,
+                  }}
+                  frameTextValueStyle={{
+                    fontSize: hasLastSalePrice ? '16px' : '10px',
+                    lineHeight: hasLastSalePrice ? '24px' : '16px',
+                  }}
+                  icon={
+                    lastSale?.orderSource ? (
+                      <ButtonIcon
+                        size={ButtonIconSize.Sm}
+                        padding={2}
+                        color={IconColor.iconMuted}
+                        onClick={() => {
+                          global.platform.openTab({
+                            url: lastSale?.orderSource as string,
+                          });
+                        }}
+                        iconName={IconName.Export}
+                        ariaLabel="redirect"
+                      />
+                    ) : undefined
+                  }
+                />
+                <NftDetailInformationFrame
+                  frameClassname="nft-details__nft-frame"
+                  title={t('highestFloorPrice')}
+                  frameTextTitleProps={{
+                    textAlign: TextAlign.Center,
+                    color: TextColor.textAlternative,
+                    variant: TextVariant.bodyMdMedium,
+                  }}
+                  frameTextTitleStyle={{
+                    fontSize: '10px',
+                    lineHeight: '16px',
+                  }}
+                  value={
+                    collection?.floorAsk?.price?.amount?.usd
+                      ? formatCurrency(
+                          `${collection?.floorAsk?.price?.amount?.usd}`,
+                          currency,
+                          getPricePrecision(
```

Comment:
`getPricePrecision` is probably not needed throughout this file, and the `precision` argument to `formatCurrency` could probably be omitted.

The default precision of `formatCurrency` uses the minimum of whatever the fiat currency is.  Like 2 for USD since $.01 is the minimum unit of a dollar.  I expect that's already correct for these values.  Higher precision was only needed for token prices like when 1 PEPE can be worth less than a penny like $0.00001082.